lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -402,6 +402,14 @@ def check_watched_item(self, job_state):
                                                     namespace=self.runner_params['k8s_namespace'])
         if len(jobs.response['items']) == 1:
             job = Job(self._pykube_api, jobs.response['items'][0])
+            # Check if job.obj['status'] is empty,
```

Comment:
Can you move this to line 431 ?
It seems like that would be more coherent.
Is this response documented anywhere ?
Is this an empty list or an empty string ?

Reply:
I too was wondering whether this needs an upstream bug report in k8s or something.

Reply:
Well, I don't think that this is really a failure, I have seen k8s filling API objects progressively as information becomes available by the API server in other cases.

Reply:
Sure, I can move to 431, but none of what happens in the middle is really needed if the list is empty, are you happy with that?